github/code-scanning.go

```diff
@@ -171,3 +181,30 @@ func (s *CodeScanningService) GetAlert(ctx context.Context, owner, repo string,
 
 	return a, resp, nil
 }
+
+// UploadSarif uploads the result of code scanning job to GitHub.
+//
+// For the parameter sarif, you must first compress your SARIF file using gzip and then translate the contents of the file into a Base64 encoding string
+// You must use an access token with the security_events scope to use this endpoint. GitHub Apps must have the security_events
+// write permission to use this endpoint.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#upload-an-analysis-as-sarif-data
+func (s *CodeScanningService) UploadSarif(ctx context.Context, owner, repo string, commitSHA string, ref string, sarif string) (*Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/code-scanning/sarifs", owner, repo)
+	results := &AnalysisResults{
+		CommitSHA: &commitSHA,
+		Ref:       &ref,
+		Sarif:     &sarif,
+	}
+	req, err := s.client.NewRequest("POST", u, results)
+	if err != nil {
+		return nil, err
+	}
+	analysis := new(AnalysisResults)
```

Comment:
Please add a blank line before this line for consistency with the rest of the repo.

Also, this is not correct... the `AnalysisResults` struct is not being passed back in the response.  Please re-read the documentation and create a new `SarifID` struct that has the `id` and `url` fields in it and return that instead.